pkg/devspace/dependency/resolver.go

```diff
@@ -130,7 +131,7 @@ func (r *resolver) resolveRecursive(ctx devspacecontext.Context, basePath, paren
 			continue
 		}
 
-		dependencyConfigPath, err := util.DownloadDependency(ctx.Context(), basePath, dependencyConfig.Source, ctx.Log())
+		dependencyConfigPath, err := util.DownloadDependency(ctx.Context(), basePath, dependencyConfig.Source, ctx.Log(), r.DownloadedIds)
```

Comment:
You might be able to skip this by doing a check similar to [this line](https://github.com/devspace-sh/devspace/pull/2769/files#diff-619d41cab5b9425e3f17b291d9018247b2ce6c2a7d9ec9d56e65a30602191bceR143).